exercises/grade-school2/canonical-data.json

```diff
@@ -0,0 +1,185 @@
+          },
+          "expected": [true, true, false, true]
+        },
+        {
+          "uuid": "c7ec1c5e-9ab7-4d3b-be5c-29f2f7a237c5",
+          "description": "Student not added to multiple grades in the roster",
+          "property": "roster",
+          "input": {
+            "students": [["Blair", 2], ["James", 2], ["James", 3], ["Paul", 3]]
+          },
+          "expected": ["Blair", "James", "Paul"]
+        },
+        {
+          "uuid": "d9af4f19-1ba1-48e7-94d0-dabda4e5aba6",
+          "description": "Students are sorted by grades in the roster",
+          "property": "roster",
+          "input": {
+            "students": [["Jim", 3], ["Peter", 2], ["Anna", 1]]
+          },
+          "expected": ["Anna", "Peter", "Jim"]
+        },
+        {
+          "uuid": "d9fb5bea-f5aa-4524-9d61-c158d8906807",
+          "description": "Students are sorted by name in the roster",
+          "property": "roster",
+          "input": {
+            "students": [["Peter", 2], ["Zoe", 2], ["Alex", 2]]
+          },
+          "expected": ["Alex", "Peter", "Zoe"]
+        },
+        {
+          "uuid": "4227a6ba-7734-4029-83bc-24b1e0bcad82",
+          "description": "Students are sorted by grades and names in the roster",
+          "property": "roster",
+          "input": {
+            "students": [["Peter", 2], ["Anna", 1], ["Barb", 1], ["Zoe", 2], ["Alex", 2], ["Jim", 3], ["Charlie", 1]]
+          },
+          "expected": ["Anna", "Barb", "Charlie", "Alex", "Peter", "Zoe", "Jim"]
+        },
+        {
+          "uuid": "c92effa5-2cbf-4261-b3bc-a7f9e75ff0b7",
+          "description": "Grade is empty if no students in the roster",
+          "property": "grade",
+          "input": {
+            "students": [],
+            "desiredGrade": 1
+          },
+          "expected": []
+        },
+        {
+          "uuid": "1e0cf06b-26e0-4526-af2d-a2e2df6a51d6",
+          "description": "Grade is empty if no students in that grade",
+          "property": "grade",
+          "input": {
+            "students": [["Peter", 2], ["Zoe", 2], ["Alex", 2], ["Jim", 3]],
+            "desiredGrade": 1
+          },
+          "expected": []
+        },
+        {
+          "uuid": "2bfc697c-adf2-4b65-8d0f-c46e085f796e",
+          "description": "Student not added to same grade more than once",
+          "property": "grade",
+          "input": {
+            "students": [["Blair", 2], ["James", 2], ["James", 2], ["Paul", 2]],
+            "desiredGrade": 2
+          },
+          "expected": ["Blair", "James", "Paul"]
+        },
+        {
+          "uuid": "66c8e141-68ab-4a04-a15a-c28bc07fe6b9",
+          "description": "Student not added to multiple grades",
+          "property": "grade",
+          "input": {
+            "students": [["Blair", 2], ["James", 2], ["James", 3], ["Paul", 3]],
+            "desiredGrade": 5
+          },
+          "expected": ["Blair", "James", "Paul"]
```

Comment:
I don't think it can be proper for grade 5 to have a nonzero number of students when no students were added to grade 5.

Reply:
Whoops, copy-pasta error